Builds/CMake/FindBoost.cmake

```diff
@@ -458,20 +458,10 @@ function(_Boost_GUESS_COMPILER_PREFIX _ret)
   elseif (GHSMULTI)
     set(_boost_COMPILER "-ghs")
   elseif("x${CMAKE_CXX_COMPILER_ID}" STREQUAL "xMSVC")
-    if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.10)
+    if(MSVC_TOOLSET_VERSION GREATER_EQUAL 141)
       set(_boost_COMPILER "-vc141;-vc140")
-    elseif (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19)
-      set(_boost_COMPILER "-vc140")
-    elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 18)
-      set(_boost_COMPILER "-vc120")
-    elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 17)
-      set(_boost_COMPILER "-vc110")
-    elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16)
-      set(_boost_COMPILER "-vc100")
-    elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 15)
-      set(_boost_COMPILER "-vc90")
-    elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 14)
-      set(_boost_COMPILER "-vc80")
+    elseif(MSVC_TOOLSET_VERSION GREATER_EQUAL 80)
+      set(_boost_COMPILER "-vc${MSVC_TOOLSET_VERSION}")
```

Comment:
I doubt a version below 18 (vc120) will compile the project. We may want to remove the attempt to set anything less than vc140 in the future.

Reply:
yeah - that's a good point, but this file really comes from kitware. We keep a local copy because we'd rather not force everyone to upgrade cmake when we update boost, but this is admittedly only marginally better.

Reply:
This file is a modified version from https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake Every time a new boost comes out we'll have to re-apply the changes (or possibly require a new version of cmake). As long as that's true, I'd like to keep the modifications to the minimum needed to let this work (@mellery451 had to make some changes to let it work on windows for cmake versions < 3.12).